src/gcp/secretManager.ts

```diff
@@ -63,123 +94,131 @@ export async function secretExists(projectId: string, name: string): Promise<boo
   };
 }
 
 export function parseSecretVersionResourceName(resourceName: string): SecretVersion {
-  const nameTokens = resourceName.split("/");
+  const tokens = resourceName.match(SECRET_VERSION_NAME_REGEX);
+  if (tokens == null) {
+    throw new FirebaseError(`Invalid secret version resource name [${resourceName}].`);
+  }
   return {
     secret: {
-      projectId: nameTokens[1],
-      name: nameTokens[3],
+      projectId: tokens[1],
+      name: tokens[2],
     },
-    versionId: nameTokens[5],
+    version: tokens[3],
   };
 }
 
 export function toSecretVersionResourceName(secretVersion: SecretVersion): string {
-  return `projects/${secretVersion.secret.projectId}/secrets/${secretVersion.secret.name}/versions/${secretVersion.versionId}`;
+  return `projects/${secretVersion.secret.projectId}/secrets/${secretVersion.secret.name}/versions/${secretVersion.version}`;
 }
 
 export async function createSecret(
   projectId: string,
   name: string,
   labels: Record<string, string>
 ): Promise<Secret> {
-  const createRes = await api.request(
-    "POST",
-    `/v1beta1/projects/${projectId}/secrets?secretId=${name}`,
+  const createRes = await client.post<CreateSecretRequest, Secret>(
+    `projects/${projectId}/secrets?secretId=${name}`,
     {
-      auth: true,
-      origin: api.secretManagerOrigin,
-      data: {
-        replication: {
-          automatic: {},
-        },
-        labels,
+      replication: {
+        automatic: {},
       },
+      labels,
     }
   );
   return parseSecretResourceName(createRes.body.name);
 }
 
-export async function addVersion(secret: Secret, payloadData: string): Promise<SecretVersion> {
-  const res = await api.request(
-    "POST",
-    `/v1beta1/projects/${secret.projectId}/secrets/${secret.name}:addVersion`,
+export async function addVersion(
+  secret: Secret,
+  payloadData: string
+): Promise<Required<SecretVersion>> {
+  const res = await client.post<AddVersionRequest, { name: string; state: SecretVersionState }>(
+    `projects/${secret.projectId}/secrets/${secret.name}:addVersion`,
     {
-      auth: true,
-      origin: api.secretManagerOrigin,
-      data: {
-        payload: {
-          data: Buffer.from(payloadData).toString("base64"),
-        },
+      payload: {
+        data: Buffer.from(payloadData).toString("base64"),
       },
     }
   );
-  const nameTokens = res.body.name.split("/");
   return {
-    secret: {
-      projectId: nameTokens[1],
-      name: nameTokens[3],
-    },
-    versionId: nameTokens[5],
+    ...parseSecretVersionResourceName(res.body.name),
+    state: res.body.state,
   };
 }
 
-export async function grantServiceAgentRole(
+export async function getIamPolicy(secret: Secret): Promise<iam.Policy> {
+  const res = await client.get<iam.Policy>(
+    `projects/${secret.projectId}/secrets/${secret.name}:getIamPolicy`
+  );
+  return res.body;
+}
+
+export async function setIamPolicyBindings(secret: Secret, bindings: iam.Binding[]): Promise<void> {
```

Comment:
nit: why "Bindings"?

Reply:
I thought there were more things than bindings in a policy (there arer, but not really mutable by the user).